pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/OfflineSegmentIntervalCheckerCommand.java

```diff
@@ -0,0 +1,149 @@
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+import org.apache.pinot.common.config.TableConfig;
+import org.apache.pinot.common.config.TableNameBuilder;
+import org.apache.pinot.common.metadata.ZKMetadataProvider;
+import org.apache.pinot.common.metadata.segment.OfflineSegmentZKMetadata;
+import org.apache.pinot.controller.util.SegmentIntervalUtils;
+import org.apache.pinot.tools.Command;
+import org.joda.time.Interval;
+import org.kohsuke.args4j.Option;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Pinot admin command to list all offline segments with invalid intervals, group by table name
+ */
+public class OfflineSegmentIntervalCheckerCommand extends AbstractBaseAdminCommand implements Command {
+  private static final Logger LOGGER = LoggerFactory.getLogger(OfflineSegmentIntervalCheckerCommand.class);
+
+  private ZKHelixAdmin _helixAdmin;
+  private ZkHelixPropertyStore<ZNRecord> _propertyStore;
+
+  @Option(name = "-zkAddress", required = true, metaVar = "<http>", usage = "Zookeeper server:port/cluster")
+  private String _zkAddress;
+
+  @Option(name = "-clusterName", required = true, metaVar = "<String>", usage = "Helix cluster name")
+  private String _clusterName;
+
+  @Option(name = "-tableNames", metaVar = "<string>", usage = "Comma separated list of tables to check for invalid segment intervals")
+  private String _tableNames;
+
+  @Option(name = "-help", help = true, aliases = {"-h", "--h", "--help"}, usage = "Print this message.")
+  private boolean _help = false;
+
+  @Override
+  public boolean getHelp() {
+    return _help;
+  }
+
+  @Override
+  public String toString() {
+    return ("OfflineSegmentIntervalChecker");
+  }
+
+  @Override
+  public String getName() {
+    return "OfflineSegmentIntervalChecker";
+  }
+
+  @Override
+  public void cleanup() {
+
+  }
+
+  @Override
+  public String description() {
+    return "Prints out offline segments with invalid time intervals";
+  }
+
+  @Override
+  public boolean execute()
+      throws Exception {
+    LOGGER.info("Executing command: " + toString());
+
+    _helixAdmin = new ZKHelixAdmin(_zkAddress);
+    _propertyStore = new ZkHelixPropertyStore<>(_zkAddress, new ZNRecordSerializer(),
+        PropertyPathConfig.getPath(PropertyType.PROPERTYSTORE, _clusterName));
+
+    List<String> offlineTables = new ArrayList<>();
+    if (StringUtils.isBlank(_tableNames)) {
+      List<String> resourcesInCluster = _helixAdmin.getResourcesInCluster(_clusterName);
+      for (String tableName : resourcesInCluster) {
+        if (TableNameBuilder.isOfflineTableResource(tableName)) {
+          offlineTables.add(tableName);
+        }
+      }
+    } else {
+      for (String tableName : _tableNames.split(",")) {
+        if (ZKMetadataProvider.getOfflineTableConfig(_propertyStore, tableName) != null) {
+          offlineTables.add(tableName);
+        }
+      }
+    }
+
+    LOGGER.info("Tables to check: {}", offlineTables);
```

Comment:
Does it print out the content of the array list or just the physical address? Or should we just print the number of offline tables to check?

Reply:
It prints contents of the array. We want the tables to be listed. This is an adhoc tool, so shouldn't matter that much

Reply:
Yeah, but the table names will be printed in line 118 though. Not sure that's duplicated or not..

Reply:
My intention was to print out the list upfront here, because some of the tables could get skipped because of the conditions just above. How about I let it stick for now, and I'll remove it later if it becomes a nuisance during analysis? :)